### PR TITLE
LW Handle final blit in SRP

### PR DESCRIPTION
**Pull request by aleks01010101**
### Purpose of this PR

Some platforms (Android) are doing a system blit when the backbuffer doesn't support a certain configuration. This PR makes LW responsible for doing this blit, removing double-blit situations.

---
### Release Notes

LWRP now handles system bliss for platforms and configurations that require this, removing a redundant blit operation.

---
### Testing status
**Katana Tests**: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?automation-tools_branch=add-platform-filter&ScriptableRenderLoop_branch=lw%2Fhandle_final_blit_in_srp&unity_branch=trunk

**Manual Tests**: What did you do?
Manually tested on a device (one for now, will test some later).

**Automated Tests**: What did you setup?
Nothing!

Any test projects to go with this to help reviewers?
Nope, just use the template, if you want to check this :)
---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low.

**Halo Effect**: None, Low, Medium, High?
Low (Android and iOS).
---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by nerites** [comment]
adding @brandon-unity3d to comment on how this might interact with his latest y-flip fixes, if at all

File: com.unity.render-pipelines.lightweight/Runtime/DefaultRendererSetup.cs [hunk 1]
```diff
@@ -264,7 +271,7 @@ public void Setup(ScriptableRenderer renderer, ref RenderingData renderingData)
                 //now blit into the final target
                 if (colorHandle != RenderTargetHandle.CameraTarget)
                 {
-                    m_FinalBlitPass.Setup(baseDescriptor, colorHandle);
+                    m_FinalBlitPass.Setup(baseDescriptor, colorHandle, Display.main.requiresSRGBBlitToBackbuffer, killAlpha);
```
[the remark this answers is not in the document]
Comment (aleks01010101) [reply com.unity.render-pipelines.lightweight/Runtime/DefaultRendererSetup.cs]: It should be, but I didn't find a good way to determine, to which display we're rendering yet.

File: com.unity.render-pipelines.lightweight/Shaders/Utils/Blit.shader [hunk 2]
```diff
@@ -45,6 +51,12 @@ Shader "Hidden/Lightweight Render Pipeline/Blit"
             half4 Fragment(Varyings input) : SV_Target
             {
                 half4 col = SAMPLE_TEXTURE2D(_BlitTex, sampler_BlitTex, input.uv);
+                #ifdef _LINEAR_TO_SRGB_CONVERSION
+                col = LinearToSRGB(col);
+                #endif
+                #ifdef _KILL_ALPHA
+                col.a = 1.0;
```
[the remark this answers is not in the document]
Comment (aleks01010101) [reply com.unity.render-pipelines.lightweight/Shaders/Utils/Blit.shader]: I'll add the stripping and check the ColorMask thing.

File: com.unity.render-pipelines.lightweight/Runtime/DefaultRendererSetup.cs [hunk 3]
```diff
@@ -163,13 +163,20 @@ public void Setup(ScriptableRenderer renderer, ref RenderingData renderingData)
                 renderer.EnqueuePass(m_ScreenSpaceShadowResolvePass);
             }
 
+            bool platformNeedsToKillAlpha = Application.platform == RuntimePlatform.IPhonePlayer ||
+                Application.platform == RuntimePlatform.Android ||
+                Application.platform == RuntimePlatform.tvOS;
+            bool killAlpha = !Graphics.preserveFramebufferAlpha && platformNeedsToKillAlpha;
```
[the remark this answers is not in the document]
Comment (aleks01010101) [reply com.unity.render-pipelines.lightweight/Runtime/DefaultRendererSetup.cs]: will do

[the remark this answers is not in the document]
Comment (aleks01010101) [reply com.unity.render-pipelines.lightweight/Runtime/DefaultRendererSetup.cs]: done

File: com.unity.render-pipelines.lightweight/Shaders/Utils/Blit.shader (hunk 2 above)
[the remark this answers is not in the document]
Comment (aleks01010101) [reply com.unity.render-pipelines.lightweight/Shaders/Utils/Blit.shader]: stripping done

File: com.unity.render-pipelines.lightweight/CHANGELOG.md [hunk 4]
```diff
@@ -11,11 +11,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Fixed
 - LWRP now respects the iOS Player setting **Force hard shadows**. When you enable this setting, hardware filtering of shadows is disabled.
 
-### Changed
+### Added
 - The LWRP package is no longer in preview.
 - LWRP built-in render passes are now internal.
 - Changed namespace from `UnityEngine.Experimental.Rendering.LightweightPipeline` to `UnityEngine.Rendering.LWRP`.
-- Changed namespace from `UnityEditor.Experimental.Rendering.LightweightPipeline` to `UnityEditor.Rendering.LWRP`.
+- LWRP now handles blits that are required by the device when rendering to the backbuffer.
```
Comment (phi-lira) [inline com.unity.render-pipelines.lightweight/CHANGELOG.md]: The merge on the changelog is was not done properly. Some things were removed, There are now two Fixed categories, etc.

Comment (Katstodian) [reply com.unity.render-pipelines.lightweight/CHANGELOG.md]: Good catch. 
@phi-lira want to merge this and then do a separate PR for the changelog, or change in this?

File: com.unity.render-pipelines.lightweight/Editor/ShaderPreprocessor.cs [hunk 5]
```diff
@@ -55,6 +57,15 @@ bool StripUnusedShader(ShaderFeatures features, Shader shader)
                 shader.name.Contains("ScreenSpaceShadows"))
                 return true;
 
+            if (shader.name.EndsWith("/Blit"))
+            {
+                bool preserveAlpha = PlayerSettings.preserveFramebufferAlpha && compilerData.platformKeywordSet.IsEnabled(BuiltinShaderDefine.SHADER_API_MOBILE);
```
[the remark this answers is not in the document]
Comment (aleks01010101) [reply com.unity.render-pipelines.lightweight/Editor/ShaderPreprocessor.cs]: They are the same. PlayerSettings is an Editor-only thing, so I exposed it in GraphicsSettings.

File: com.unity.render-pipelines.lightweight/Editor/ShaderPreprocessor.cs [hunk 6]
```diff
@@ -55,6 +57,15 @@ bool StripUnusedShader(ShaderFeatures features, Shader shader)
                 shader.name.Contains("ScreenSpaceShadows"))
                 return true;
 
+            if (shader.name.EndsWith("/Blit"))
```
[the remark this answers is not in the document]
Comment (aleks01010101) [reply com.unity.render-pipelines.lightweight/Editor/ShaderPreprocessor.cs]: Will fix this

File: com.unity.render-pipelines.lightweight/Shaders/Utils/Blit.shader [hunk 7]
```diff
@@ -19,7 +19,13 @@ Shader "Hidden/Lightweight Render Pipeline/Blit"
             #pragma vertex Vertex
             #pragma fragment Fragment
 
+            #pragma multi_compile _ _LINEAR_TO_SRGB_CONVERSION
+            #pragma multi_compile _ _KILL_ALPHA
+
             #include "Packages/com.unity.render-pipelines.lightweight/ShaderLibrary/Core.hlsl"
+            #ifdef _LINEAR_TO_SRGB_CONVERSION
```
[the remark this answers is not in the document]
Comment (aleks01010101) [reply com.unity.render-pipelines.lightweight/Shaders/Utils/Blit.shader]: I think it's only the analysis stage that doesn't.

**Pull request merged**